### PR TITLE
Fix custom component CLI on main/5.0

### PR DESCRIPTION
**Pull request by freddyaboulton**
## Description

The change to the `package.json` means that some code in the custom component cli broke. The following changes are needed to fix. `dev` should work but `build` is broken due to the svelte regex. Leaving this PR as draft. @pngwn maybe you can push your changes here?

## 🎯 PRs Should Target Issues

Before your create a PR, please check to see if there is [an existing issue](https://github.com/gradio-app/gradio/issues) for this change. If not, please create an issue before you create this PR, unless the fix is very small. 

Not adhering to this guideline will result in the PR being closed. 

## Tests

1. PRs will only be merged if tests pass on CI. To run the tests locally, please set up [your Gradio environment locally](https://github.com/gradio-app/gradio/blob/main/CONTRIBUTING.md) and run the tests: `bash scripts/run_all_tests.sh`

2. You may need to run the linters: `bash scripts/format_backend.sh` and `bash scripts/format_frontend.sh`

### CODE REVIEW

File: guides/08_custom-components/06_frequently-asked-questions.md [hunk 1]
```diff
@@ -1,7 +1,14 @@
 # Frequently Asked Questions
 
 ## What do I need to install before using Custom Components?
-Before using Custom Components, make sure you have Python 3.8+, Node.js v16.14+, npm 9+, and Gradio 4.0+ installed.
+Before using Custom Components, make sure you have Python 3.10+, Node.js v16.14+, npm 9+, and Gradio 4.0+ installed.
```
Comment (pngwn) [inline guides/08_custom-components/06_frequently-asked-questions.md:4]: lets bump this to node 18, i don't think it is needed but better we do it now than later and 16 is pretty old now (22 is current).

File: guides/08_custom-components/06_frequently-asked-questions.md [hunk 2]
```diff
@@ -1,7 +1,14 @@
...
+
+## Are custom components compatible between Gradio 4.0 and 5.0?
+
+Custom components built with Gradio 5.0 should be compatible with Gradio 4.0. If you built your custom component in Gradio 4.0 you will have to rebuild your component to be compatible with Gradio 5.0. Simply follow these steps:
+1. Update the `@gradio/preview` package. Cd into the `frontend` directory and run `npm update`.
```
Comment (pngwn) [inline guides/08_custom-components/06_frequently-asked-questions.md:9]:
```suggestion
1. Update the `@gradio/preview` package. `cd` into the `frontend` directory and run `npm update`.
```

File: js/app/src/routes/svelte_init.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+			o[key] = o["SvelteComponent"];
+		} else {
+			//@ts-ignore
+			o[key] = svelte[key];
+		}
+	}
+	window.__gradio__svelte__internal = o;
+	window.__gradio__svelte__internal["globals"] = {};
+	window.globals = window;
+	// globalThis.globals = {};
```
Comment (pngwn) [inline js/app/src/routes/svelte_init.ts:20]: remove

File: gradio/routes.py [hunk 4]
```diff
@@ -586,10 +596,18 @@ def static_resource(path: str):
...
+            environment: Literal["client", "server"],
+            type: str,
+            file_name: str,
+            req: fastapi.Request,
         ):
+            if environment not in ["client", "server"]:
+                raise HTTPException(
+                    status_code=404, detail="Environment not supported."
+                )
             config = app.get_blocks().config
```
Comment (pngwn) [inline gradio/routes.py:611]: Do we want to remove this for now or leave it in?

**Review by pngwn — APPROVED** [review]
Looks great! Thanks for taking care of this @freddyaboulton!

File: gradio/routes.py (hunk 4 above)
Comment (freddyaboulton) [reply gradio/routes.py:611]: Let's leave it in so we don't have to make changes to the route later. But `environment="sever"` basically does nothing now.

**Comment by freddyaboulton** [comment]
Thank you @pngwn !!

**Pull request merged**